src/main/java/org/openrewrite/kotlin/internal/KotlinTreeParserVisitor.java

```diff
@@ -1833,9 +1821,9 @@ public J visitCallExpression(KtCallExpression expression, ExecutionContext data)
         }
         PsiElementAssociations.ExpressionType type = psiElementAssociations.getCallType(expression);
         if (type == PsiElementAssociations.ExpressionType.CONSTRUCTOR) {
+            JavaType.Method mt = methodInvocationType(expression);
```

Comment:
Note:

PSI to FIR association does not guarantee that types are correct via `visitSimpleNameExpression`.
Rather than return `null` by default, I allowed the associated FIR element to set the type. ( which may be a mistake )

So, when applicable, we'll need to compare types from the `JavaParser` to update types on names so that type utilities will work appropriately. Once we notice a pattern, helper methods will help us a lot.

Example:
`val foo = Triple(1, 2, 3)`
The identifier `Triple` is associated with a resolved name reference of a `FirConstructor`.

In Java, a `J.NewClass` has a type of JavaType$Method like `kotlin.Triple<kotlin.Int, kotlin.Int, kotlin.Int>{name=<constructor>,return=kotlin.Triple<kotlin.Int, kotlin.Int, kotlin.Int>,parameters=[kotlin.Int,kotlin.Int,kotlin.Int]}`
and the name would be the class type `kotlin.Triple<kotlin.Int, kotlin.Int, kotlin.Int>` ( I don't know if it would be parameterized.

If the constructor were parameterized, I.E., `ArrayList<String>()`
The `PSI` will likely create a `JavaType$Parameterized` or `JavaType.Method` for the name `ArrayList` due to the associated FIR, and the PSI-parser would create a `J.Parameterized` due to type arguments.

In this case:
```
J.NewClass(
name => J.Parameterized(
                    clazz => JavaType.Class => `java.util.ArrayList`
                    type => JavaType.Parameterized => `java.util.ArrayList<java.lang.String>`
                )
type => JavaType.MethodType => `java.util.ArrayList{name=<constructor>,returnType ...`
)
```

This change is an attempt at setting the appropriate types.